src/psyclone/dynamo0p3.py

```diff
@@ -8723,6 +8680,26 @@ def intrinsic_type(self):
         '''
         return self._intrinsic_type
 
+    @property
+    def data_type(self):
```

Comment:
This method and the next should really have a base class implementation as they are not LFRic-specific, even if they are not used by other subclasses. The problem is that this functionality might end up being implemented in a different way in the other subclasses. However, seeing as we're planning to restructure everything at some point I think it is OK for the moment.

Reply:
I was thinking about whether to add it directly into the base class, as `intrinsic_type` is defined in the base class. However, what stopped me from doing that was that for APIs that do not use data objects `intrinsic_type` and `data_type` are essentially the same (e.g. a `real`-valued array). For LFRic, `data_type` indicates the type of a data object (e.g. `operator_type`, `integer_field_type`) and `intrinsic_type` is the Fortran datatype of its data (e.g, `real`, `integer`). What do you think?